recovery/bootselectiondialog.cpp

```diff
@@ -157,13 +157,33 @@ void BootSelectionDialog::accept()
     QSettings settings("/settings/noobs.conf", QSettings::IniFormat, this);
     QVariantMap m = item->data(Qt::UserRole).toMap();
     QByteArray partition = m.value("partitions").toList().first().toByteArray();
-    QRegExp partnrRx("([0-9]+)$");
-    if (partnrRx.indexIn(partition) == -1)
+
+    int partitionNr;
+    QRegExp parttype("^PARTUUID");
+    if (parttype.indexIn(partition) == -1)
```

Comment:
You could probably use http://doc.qt.io/archives/qt-4.8/qstring.html#startsWith here instead?